charms/istio-pilot/src/charm.py

```diff
@@ -431,10 +431,18 @@ def _get_ingress_data(self, event) -> dict:
         # Get all route data from the ingress interface
         routes = get_routes_from_ingress_interface(ingress_interface, self.app)
 
-        # The app-level data is still visible on a broken relation, but we
+        # The app-level data may still be visible on a broken relation, but we
         # shouldn't be keeping the VirtualService for that related app.
         if isinstance(event, RelationBrokenEvent) and event.relation.name == INGRESS_RELATION_NAME:
-            routes.pop((event.relation, event.app))
+            if event.app is None:
+                self.log.info(
```

Comment:
Should this be an ERROR log level and not INFO?

Reply:
We want just to log the incident and move on.